src/app/backing/components/BuildersSpotlight/BuildersSpotlight.tsx

```diff
@@ -18,59 +21,59 @@ interface BuildersSpotlightProps {
 
 export const BuildersSpotlight: FC<BuildersSpotlightProps> = ({ rewardsData }) => {
   const {
-    data: builders,
-    isLoading: buildersLoading,
-    error: buildersError,
-  } = useGetBuildersByState<RequiredBuilder>()
+    state: {
+      allocations,
+      builders,
+      isContextLoading,
+      getBuilder,
+      backer: { amountToAllocate: totalOnchainAllocation },
+    },
+  } = useContext(AllocationsContext)
 
-  const buildersToShow = builders.filter(({ stateFlags }) => isBuilderRewardable(stateFlags))
+  const hasAllocations = totalOnchainAllocation > 0n
 
-  const {
-    data: buildersWithBackerRewards,
-    isLoading,
-    error: buildersWithBackerRewardsError,
-  } = useGetBackersRewardPercentage(buildersToShow.map(({ address }) => address))
-
-  const router = useRouter()
-  const shuffledBuilders = useShuffledArray<Builder>(buildersToShow)
+  const randomBuilders = useShuffledArray<Builder>(Object.values(builders)).filter(({ stateFlags }) =>
+    isBuilderRewardable(stateFlags),
+  )
 
-  useHandleErrors({
-    error: buildersWithBackerRewardsError ?? buildersError,
-    title: 'Error loading builders',
+  const allocatedBuilders = Object.entries(allocations).map(([key, value]) => {
+    const builder = getBuilder(key as Address)!
+    return {
+      ...builder,
+    }
```

Comment:
can this be simplified like so?

```suggestion
  const allocatedBuilders = Object.keys(allocations).map(key => getBuilder(key as Address)!)
```